### PR TITLE
[5.7][🍒] Back out "Fix jump-to-null issue for ObjC async calls."

### PR DESCRIPTION
**Pull request by kavon**
cherry-pick of https://github.com/apple/swift/pull/58474

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci please test

**Review by DougGregor — APPROVED** [review]

File: lib/SIL/IR/SILInstruction.cpp [hunk 1]
```diff
@@ -1646,6 +1646,10 @@ bool SILInstruction::maySuspend() const {
   // await_async_continuation always suspends the current task.
   if (isa<AwaitAsyncContinuationInst>(this))
     return true;
+
+  // hop_to_executor also may cause a suspension
+  if (isa<HopToExecutorInst>(this))
+    return true;
```
Comment (rjmccall) [inline lib/SIL/IR/SILInstruction.cpp:1652]: Haha, oops

File: lib/SILGen/SILGenApply.cpp [hunk 2]
```diff
@@ -4427,6 +4427,27 @@ class FixLifetimeDestroyCleanup : public Cleanup {
...
+  void emit(SILGenFunction &SGF, CleanupLocation l,
+            ForUnwind_t forUnwind) override {
+    breadcrumb.emit(SGF, l);
+  }
+
+  void dump(SILGenFunction &SGF) const override {
+#ifndef NDEBUG
+    llvm::errs() << "EmitBreadcrumbCleanup "
+                 << "State:" << getState()
+                 << "NeedsEmit:" << breadcrumb.needsEmit();
```
Comment (rjmccall) [inline lib/SILGen/SILGenApply.cpp:4447]: Not the most important thing, but there's a missing space here before `NeedsEmit:`

Comment (kavon) [reply lib/SILGen/SILGenApply.cpp:4447]: it's also suppose to have newlines and such, now that I've been debugging the clean-ups and relying on it 😄

**Pull request merged**